File: src/main/scala/org/drugis/mtc/yadas/SuccessProbabilityArgumentMaker.scala
package org.drugis.mtc.yadas

import org.drugis.mtc._
import gov.lanl.yadas.ArgumentMaker

/**
 * ArgumentMaker for individual treatment success probabilities within studies.
 * p_i,k = ilogit(theta_i,k) ; theta_i,k = mu_i + delta_i,b(i),k
 */
class SuccessProbabilityArgumentMaker(
		override val model: NetworkModel[DichotomousMeasurement],
		override val sIdx: Int,
		override val dIdx: Int,
		override val study: Study[DichotomousMeasurement])
extends ArgumentMaker with ThetaMaker[DichotomousMeasurement] {
	/**
	 * Calculate "the argument": an array of succes-probabilities, one for
	 * each study-arm.
	 * data[sIdx] should contain study baseline means, one per study
	 * data[dIdx] should contain relative effects, one for each non-baseline
	 */
	def getArgument(data: Array[Array[Double]]): Array[Double] = {
		Array.make(0, 0.0) ++ {
			for {t <- treatments} yield prob(t, data)
		}
	}

	private def ilogit(x: Double): Double = 1 / (1 + Math.exp(-x))

	private def prob(t: Treatment, data: Array[Array[Double]])
	: Double = {
		ilogit(theta(t, data))
	}
}

File: mtc-data/src/main/java/org/drugis/mtc/yadas/SuccessProbabilityArgumentMaker.java
package org.drugis.mtc.yadas;

import java.util.List;

import org.drugis.mtc.model.Treatment;
import static org.drugis.common.stat.Statistics.ilogit;

import edu.uci.ics.jung.graph.util.Pair;

/**
 * ArgumentMaker for individual treatment success probabilities within studies.
 * p_i,k = ilogit(theta_i,k) ; theta_i,k = mu_i + delta_i,b(i),k
 */
public class SuccessProbabilityArgumentMaker extends ThetaArgumentMaker {
	public SuccessProbabilityArgumentMaker(List<Treatment> treatments, List<List<Pair<Treatment>>> studyPmtz, int muIdx, int deltaIdx) {
		super(treatments, studyPmtz, muIdx, deltaIdx);
	}
	
	@Override
	protected double theta(int tIdx, double[][] data) {
		return ilogit(super.theta(tIdx, data));
	}
}